packages/editor-ui/src/views/NodeView.vue

```diff
@@ -3228,12 +3227,13 @@ export default defineComponent({
 
 				await this.addNodes([newNodeData], [], true);
 
-				const pinData = this.workflowsStore.pinDataByNodeName(nodeName);
-				if (pinData) {
-					this.workflowsStore.pinData({
-						node: newNodeData,
-						data: pinData,
-					});
+				const pinDataForNode = this.workflowsStore.pinDataByNodeName(
+					nodeName,
+				) as INodeExecutionData[];
```

Comment:
I'd expect `pinDataByNodeName` to always return this type? Or `[]` if  none.

Reply:
Fixed, good point!